runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +111,43 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Removes a node from the layout
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
+        Layout newLayout = builder.removeLayoutServer(endpoint)
+                .removeSequencerServer(endpoint)
+                .removeLogunitServer(endpoint)
+                .removeUnResponsiveServer(endpoint)
+                .setEpoch(currentLayout.getEpoch() + 1)
+                .build();
+
+        // Seal after constructing the layout, so that the system
+        // isn't blocked if the builder throws an exception
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        newLayout.setRuntime(runtime);
+        attemptConsensus(newLayout);
+
+        runtime.invalidateLayout();
+
+        try {
+            reconfigureSequencerServers(currentLayout, newLayout, false);
+        } catch (InterruptedException ie) {
+            log.error("removeNode: Bootstrapping sequencer failed due to exception : ", ie);
+            throw new UnrecoverableCorfuInterruptedError(ie);
+        } catch (ExecutionException ee) {
+            log.error("removeNode: Bootstrapping sequencer failed due to exception : ", ee);
```

Comment:
Sorry, not handling this doesn't "fix" the problem. At the end of removeNode's execution, what are my guarantees? Is the node removed or not? If the node is guaranteed to be removed, this needs to throw an exception. If not, then how the client should handle it better be documented.

Reply:
It actually does.

This method doesn't provide guarantees. I added more documentation.